src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -113,6 +123,32 @@ public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
 
+            if (this.ParameterSetName.Equals(MetadataParameterSet, StringComparison.Ordinal))
+            {
+                MetadataResponse metadataEndpoints = null;
+                ResourceManagerEndpoint = ARMEndpoint;
+                try
+                {
+                    metadataEndpoints = EnvironmentHelper.RetrieveMetaDataEndpoints(ResourceManagerEndpoint).Result;
+                    string domain = EnvironmentHelper.RetrieveDomain(metadataEndpoints.PortalEndpoint);
+                    ActiveDirectoryEndpoint = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/') + '/';
+                    ActiveDirectoryServiceEndpointResourceId = metadataEndpoints.authentication.Audiences[0];
+                    GalleryEndpoint = metadataEndpoints.GalleryEndpoint;
+                    GraphEndpoint = metadataEndpoints.GraphEndpoint;
+                    AzureKeyVaultDnsSuffix = string.Format("vault.{0}", domain).ToLowerInvariant();
+                    AzureKeyVaultServiceEndpointResourceId = string.Format("https://vault.{0}", domain).ToLowerInvariant();
+                    StorageEndpoint = domain;
+                    EnableAdfsAuthentication = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/').EndsWith("/adfs", System.StringComparison.OrdinalIgnoreCase);
+                }
+                catch (AggregateException ae)
+                {
+                    foreach (Exception ex in ae.Flatten().InnerExceptions)
+                    {
+                        throw ex;
```

Comment:
this is only going to throw one exception.  WIll leave for now, but you shoudlk throw the aggregate if there is more than one exception